client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -122,6 +122,17 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 
 	const [ isSticky, setIsSticky ] = useState( false );
 
+	const hasTrackedView = useRef( false );
+	useEffect( () => {
+		if ( showGeneratedDesigns && ! hasTrackedView.current ) {
+			hasTrackedView.current = true;
+			recordTracksEvent( 'calypso_signup_generated_design_picker_view', {
+				vertical_id: siteVerticalId,
+				generated_designs: generatedDesigns?.map( ( design ) => design.slug ).join( ',' ),
```

Comment:
[nit] I think `design_slugs` is sufficient and clearer

- no need `generated_` as the event itself already says "generated" (`calypso_signup_generated_design_picker_view`)
- need `_slugs` prefix as we're taking the slugs of the designs